File: problem22/Solution.java
import java.util.Arrays;

class Solution {
    public static boolean closeStrings(String word1, String word2) {
        if(word1.length()!=word2.length())
			return false;
        
		int w1[]=new int[26];
        int w2[]=new int[26];
        
		for(char ch:word1.toCharArray())
            w1[ch-'a']++;

        for(char ch:word2.toCharArray()){
            if(w1[ch-'a']==0) return false;
            w2[ch-'a']++;
        }

        Arrays.sort(w1);
        Arrays.sort(w2);
        return Arrays.equals(w1,w2);
    }
	public static void main(String[] args) {
		String word1 = "cabbba", word2 = "caabbb";
		System.out.println(closeStrings(word1, word2));
	}
}

File: problem22/Solution.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2):
            return False
        freq1 = [0] * 26
        freq2 = [0] * 26
        
        for c1, c2 in zip(word1, word2):
            freq1[ord(c1) - 97] += 1
            freq2[ord(c2) - 97] += 1
        
        for i in range(26):
            if (freq1[i] == 0 and freq2[i] != 0) or (freq1[i] != 0 and freq2[i] == 0):
                return False
        freq1.sort()
        freq2.sort()
        return freq1 == freq2